### ISSUE TITLE
Error while printing ROM theory.

### ISSUE THREAD

**Issue by MM45** [EasyCrypt/easycrypt#51]
Running:
```
require ROM.
print ROM.
```
causes EasyCrypt to freeze. After interrupting the process, the following is printed:
```
* In [theories]:

[error-0-0]unknown symbol: Top.ROM.Lazy.LRO./o

abstract theory ROM.
  type in_t.
  
  type out_t.
  
  op dout : Top.ROM.in_t -> Top.ROM.out_t distr.
  
  module type POracle = {
    proc o(x : Top.ROM.in_t) : Top.ROM.out_t {}
  }
  
  type d_in_t.
  
  type d_out_t.
  
  module type Distinguisher(H : Top.ROM.POracle) = {
    proc run (Top.ROM.d_in_t) : Top.ROM.d_out_t {H.o}
  }
  
  module type Oracle = {
    proc init() : unit {}
    
    proc o(x : Top.ROM.in_t) : Top.ROM.out_t {}
  }
  
  module Exp(H : Top.ROM.Oracle, D : Top.ROM.Distinguisher) = {
    proc main(x : Top.ROM.d_in_t) : Top.ROM.d_out_t = {
      var r : Top.ROM.d_out_t;
      
      H.init();
      Top.ROM.Exp(H, D).main.r <@ D(H).run(Top.ROM.Exp(H, D).main.x);
      
      return Top.ROM.Exp(H, D).main.r;
    }
  }.
  
  theory Lazy.
    module LRO = {
      var m : (Top.ROM.in_t, Top.ROM.out_t) SmtMap.fmap
      
      proc init() : unit = {
        Top.ROM.Lazy.LRO.m <- SmtMap.empty;
      }
      
      proc o(x : Top.ROM.in_t) : Top.ROM.out_t = {
        var r : Top.ROM.out_t;
        
        Top.ROM.Lazy.LRO.o.r <$ (dout Top.ROM.Lazy.LRO.o.x)%Top.ROM;
        if ((Top.ROM.Lazy.LRO.o.x \notin Top.ROM.Lazy.LRO.m)%SmtMap) 
          Top.ROM.Lazy.LRO.m.[Top.ROM.Lazy.LRO.o.x] <- Top.ROM.Lazy.LRO.o.r;
        
        return oget (Top.ROM.Lazy.LRO.m.[Top.ROM.Lazy.LRO.o.x])%SmtMap;
      }
    }.
    
    lemma LRO_init_ll: islossless Top.ROM.Lazy.LRO.init.
    
    lemma LRO_o_ll:
      (forall (x : Top.ROM.in_t), is_lossless ((dout x))%Top.ROM) =>
      islossless Top.ROM.Lazy.LRO.o.
    
    lemma LRO_initE:
      hoare[ Top.ROM.Lazy.LRO.init :
              true ==> Top.ROM.Lazy.LRO.m = SmtMap.empty].
    
    lemma LRO_oE:
      forall (M : (Top.ROM.in_t, Top.ROM.out_t) SmtMap.fmap)
        (x' : Top.ROM.in_t),
        hoare[ Top.ROM.Lazy.LRO.o : Top.ROM.Lazy.LRO.m = M /\ [2|check]>
interrupted
```

**Comment by chdoc** [comment]
@strub @bgregoir This issue is still around and affects a lot more theories than `ROM`. For instance, it also affects `PROM` and `SDist`. On the other hand, `Hybrid` and `RealSeries` are examples of theories that print fine. 

The error occurs also when checking files from the command line (in this case easycrypt terminates in the middle of printing statements without even printing a newline). 

> GIT hash: f66167f7

**Comment by strub** [comment]
I know how to fix this. This is something we should have fixed for the release.

**Comment by chdoc** [comment]
@strub I just noticed that this issue is still around, and pretty annoying.

**Comment by oskgo** [comment]
Printing an uninstantiated `LorR` also freezes.

**Comment by strub** [comment]
This bug has been fixed by 009198b24774ebd46fd0e41179376fbd71d97883
